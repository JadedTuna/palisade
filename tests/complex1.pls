in {
    high salaries[5]: int;
    low size: int;
}
out {
    low avg: int;
}

total := 0;
index := 0;

while (index < size) {
    total = total + salaries[index];
    index = index + 1;
}

avg = declassify total / size;
