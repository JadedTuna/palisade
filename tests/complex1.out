[1;34mnote: [0mlabel of [1;34mtotal[0m set to [1;93mhigh[0m
  13 |     total [1;34m=[0m total + salaries[index];
       [1;34m~~~~~~~~~~[0m[1;34m^[0m
[[1;32m OK [0m] [1;93mlow [0m [1;34mavg[0m is [1;93mlow[0m
